### DISCUSSION TITLE
Make HTTPRequest support a TimeSpan-based Timeout property instead of int for C#

### DISCUSSION THREAD

**Discussion by c-schembri** [godotengine/godot-proposals#4419] [Scripting]
Currently the `HTTPRequest` `Timeout` property is an `int`, which means a lack of precision. C# also has a `TimeSpan` `struct` which allows using whatever measurement the user decides. Internally, the timeout is a `double` so I don't see why `Timeout` is an `int` in the first place.

**Comment by Calinou** [comment]
In the `master` branch, `timeout` in HTTPRequest is already exposed as a float in the scripting API: https://github.com/godotengine/godot/blob/f4b0c7a1ea8d86c1dfd96478ca12ad1360903d9d/scene/main/http_request.cpp#L615

However, in the `3.x` branch, it's exposed as an integer: https://github.com/godotengine/godot/blob/3ba980379dfb72f5d238640835d8e77c76cf3c99/scene/main/http_request.cpp#L529

Changing this doesn't break compatibility with existing GDScript code, but it will break compatibility with existing GDNative plugins. This means the change can be considered for the future 3.x minor release, but not for a patch release.

Also, if I'm not mistaken, Godot doesn't expose time properties as TimeSpans in C# anywhere yet, so I'd vouch for sticking to a float here.

**Edit:** Pull request opened: https://github.com/godotengine/godot/pull/60402

**Comment by c-schembri** [comment]
I checked the master branch and saw that it was internally a `double` but didn't realise that had been changed in Godot 4--hence my confusion here. Thanks for your work. Your PR looks good to me.
